### PR TITLE
NPA-2388 Add Sandbox Environment 

### PR DESCRIPTION
**Pull request by JackPlowman**
## Summary
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature

## Description

Adds a sandbox API using Flask connected to NHS nonprod apigee. The sandbox api has build tools for local and running in docker, GitHub action to validate the sandbox and even documentation in the `sandbox/README.md`

I've also added a couple of workflows which check newly added dependencies and python code security (GitHub CodeQL). These tools can be used due to the repository being open source.

NOTE: the **OpenAPI spec has been changed!** This is to add example identifier and patient:identifier variables.

## Reviews Required
* [x] Dev
* [ ] Test
* [x] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The contents of this repository are protected by Crown Copyright (C).
 ### Requirements
 * make
 * nodejs + npm/yarn
-* [poetry](https://github.com/python-poetry/poetry)
+* Python 3.8 + [poetry](https://github.com/python-poetry/poetry)
```
Comment (joanne-doyle7-NHSD) [inline README.md:28]: Out of curiosity - do we use anything introduced in v3.9 that would cause us to need to bump this?

Comment (JackPlowman) [reply README.md:28]: I would like the python version to be 3.12 but I couldn't do it easily due to the common pipelines provided by APIM using Python 3.8 -> https://github.com/NHSDigital/api-management-utils/tree/master

Comment (JackPlowman) [reply README.md:28]: I don't think we need to bump the version yet due to Python 3.8 being currently supported and everything works with Python 3.8

**Review by joanne-doyle7-NHSD — APPROVED** [review]
LGTM 👍

**Pull request merged**